python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -449,25 +449,34 @@ def _get_nodes_allowed_to_terminate(
                 nodes_allowed_to_terminate[node_id] = False
         return nodes_allowed_to_terminate
 
-    def _keep_min_worker_of_node_type(
+    def _keep_worker_of_node_type(
             self, node_id: NodeID,
             node_type_counts: Dict[NodeType, int]) -> bool:
-        """Returns if workers of node_type can be terminated.
-        The worker cannot be terminated to respect min_workers constraint.
+        """Determines if a worker should be kept based on the min_workers
+        constraint of the worker's node_type.
 
-        Receives the counters of running nodes so far and determines if idle
-        node_id should be terminated or not. It also updates the counters
-        (node_type_counts), which is returned by reference.
+        Returns True exactly when both of the following hold:
+        (a) The worker's node_type is present among the keys of the current
+            config's available_node_types dict.
+        (b) Deleting the node would violate the min_workers constraint for that
+            worker's node_type.
+
+        Also updates the counter dict (node_type_counts), which is passed in by
+        reference.
```

Comment:
In the future I think it would be great if we just returned a new counts dict (as opposed to mutating the dict).

Reply:
I'll just make that change now, else this will be forgotten forever.

Reply:
That's a pretty good point--thanks!